RIOT-OS/RIOT
# cpu/esp32: Remove dependency on module libc_gettimeofday

### Contribution description

This PR adds a simple implementation of `_gettimeofday_r` based on the microsecond system timer (`system_get_time_64()`) without occupying a `TIMER_DEV.

Since PR #18027, the ESP32 port uses the module `libc_gettimeofday` by default which enables implicitly modules `xtimer` and `ztimer_usec` which in turn occupies `TIMER_DEV(0)`.

The changes in this PR
- allow to remove this dependency on module `libc_gettimeofday`.
- fix the execution error in `tests/periph_timer` due to `TIMER_DEV(0)` is ocu

In this implementation, the the microsecond system timer is realized on top of a peripheral timer. In future implementations of the ESP32 port, the microsecond hardware system timer will be used for that purpose.

### Testing procedure

1. Check the used modules list:
   ```
    BOARD=esp32-wroom-32 make -C tests/shell info-modules
   ```
   Without this PR the module list is:
   ```
   ...
   
   tsrb
   xtensa 
   xtimer
   ztimer
   ztimer64
   ztimer64_init
   ztimer64_usec
   ztimer64_xtimer_compat
   ztimer_convert
   ztimer_convert_frac
   ztimer_convert_shift
   ztimer_core
   ztimer_extend
   ztimer_init
   ztimer_periph_timer
   ztimer_usec
   ztimer_xtimer_compat
   ```
   With this PR the module list should be:
   ```
   ...
   tsrb
   xtensa
   ```
2. Flash an test
   ```
    BOARD=esp32-wroom-32 make -C tests/periph_timer`
   ```
    Without this PR the output is:
    ```
   Test for peripheral TIMERs
   Available timers: 3
   
   Testing TIMER_0:
   TIMER_0: ERROR on initialization - skipping
   ...
   TEST FAILED
   ```
   With this PR the output should be:
   ```
   Test for peripheral TIMERs
   Available timers: 3
   
   Testing TIMER_0:
   TIMER_0: initialization successful
   ...
   TEST SUCCEEDED
   ```


### Issues/PRs references

Found in test runs in CI for #16833
